### PR TITLE
Mini player redesign

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #

This PR updates the design of the mini-player with the following
 - Transparent margins around the player
 - Rounded corners 
 - Improved button sizes.
 - Shadows

A small note regarding shadow on dark mode, the shadow is always black independently of the theme so there is no noticeable shadows. This is by design defined by @david-gonzalez-a8c 

| Light | Dark | Rose | Classic | Indigo |
| - | - | - | - | - |
| Simulator Screenshot - iPhone SE (3rd generation) - 2024-04-24 at 18 14 12 | Simulator Screenshot - iPhone SE (3rd generation) - 2024-04-24 at 18 08 17 | Simulator Screenshot - iPhone SE (3rd generation) - 2024-04-24 at 18 08 31 | Simulator Screenshot - iPhone SE (3rd generation) - 2024-04-24 at 18 08 43 |  Simulator Screenshot - iPhone SE (3rd generation) - 2024-04-24 at 18 09 00 |

https://github.com/Automattic/pocket-casts-ios/assets/651601/84a4f568-56cb-4806-80ce-ab3e75270916

## To test

 - Start the app
 - Start to play an item and display the mini-player
 - Check the looks on different themes
 - Check the transitions from mini-player to full player by pressing on it

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by CookieyedCodes** [comment]
Care to share what it looks like?

**Comment by SergioEstevao** [comment]
@CookieyedCodes this is still a draft PR but I added a screenshot on the description to give you an idea where we are heading to.

**Comment by CookieyedCodes** [comment]
Ah fair enough 😉, make sure the long press artwork functionality remains 😉, most don't know about that :)

File: podcasts/MiniPlayerViewController.swift [hunk 1]
```diff
@@ -39,13 +40,20 @@ class MiniPlayerViewController: SimpleNotificationsViewController {
...
+        view.isHidden = false
 
+        setupCornersAndShadow()
         addUINotificationObservers()
         playbackStateDidChange()
         themeChanged()
     }
 
+    private func setupCornersAndShadow() {
+        let cornerRadius = CGFloat(12)
```
Comment (bjtitus) [inline podcasts/MiniPlayerViewController.swift]: Should this be defined in a Constants enum/struct in the View Controller?

**Comment by SergioEstevao** [comment]
> make sure the long press artwork functionality remains
@CookieyedCodes all good! I just tested it and it remains functional!

**Comment by rviljoen** [comment]
@SergioEstevao Had a look at the redesign and it looks really good! The only bit that feels a bit weird is the small space between the player and the tab bar, especially when scrolling. Maybe this section should be blurred, or empty? I'm sure @david-gonzalez-a8c will have some thoughts as well.

image

**Review by bjtitus — APPROVED** [review]
Left one very minor nit but code looks good and the player works well!

The only real feedback I have on the design is that the playback bar looks a lot like a shadow to me for many podcasts. It took me a second to realize it wasn't an additional shadow underneath:

CleanShot 2024-04-25 at 21 42 41@2x

CleanShot 2024-04-25 at 21 41 41@2x

File: podcasts/MiniPlayerViewController.swift (hunk 1 above)
Comment (SergioEstevao) [reply podcasts/MiniPlayerViewController.swift]: Good point I will address it.

**Comment by SergioEstevao** [comment]
> The only real feedback I have on the design is that the playback bar looks a lot like a shadow to me for many podcasts. It took me a second to realize it wasn't an additional shadow underneath:

After talking with @david-gonzalez-a8c we decided to rotate the new shadow to the top of the mini-player bar, so this effect is reduced.

**Comment by david-gonzalez-a8c** [comment]
> Maybe this section should be blurred, or empty? I'm sure @david-gonzalez-a8c will have some thoughts as well.

Good point, @rviljoen. Thank you for taking a look. I can see that being distracting.

I've added to [the specs](https://www.figma.com/file/G60UqxPhTynbalo0Vfr1NC/Playground?type=design&node-id=1074%3A8085&mode=design&t=4uaOfOo3mFGrqCyE-1) a transparent layer behind the mini player, going from $primary-ui-01 (100%) to $primary-ui-01 (0%) opacity. Can we please add that and see how it works?

image

**Comment by rviljoen** [comment]
@SergioEstevao @david-gonzalez-a8c The gradient looks awesome! One small nit is to remove the gradient from the scroll bar:

image

This should be fixed though once https://github.com/Automattic/pocket-casts-ios/issues/1072 is resolved.

**Comment by SergioEstevao** [comment]
@rviljoen thanks for the heads up regarding the scrollbar I will address that on a separate PR for #1072.

**Review by bjtitus — APPROVED** [review]
👍 Everything looks good!

**Pull request merged**